### ISSUE TITLE
Mode API

### ISSUE THREAD

**Issue by morris** [morris/vinyl-ftp#6]
#5 "Then, what is the utility of having a remoteFolder option for mode?"

Good question - It's used in combination with the relative paths of incoming vinyl files. Not sure whether I could remove it and just use the absolute paths.

However I think the best solution would be an API like "conn.mode( globs, mode )" that just globs and chmods in one go. The gulp/stream way does not have any benefits here. Will keep this issue open for some time and then decide what to do with it.

**Comment by pyrsmk** [comment]
Are `conn.delete` and `conn.rmdir` concerned too?

**Comment by morris** [comment]
Well, kind of - currently they are regular callback-based methods, no streams involved, but they don't support globs (yet). RMDIR does work recursively though, so there's no pressure on that end.

**Comment by pyrsmk** [comment]
Hum, ok. I assume that we can't glob remote files easily?
Anyway, +1 for the `conn.mode(globs, mode)` API!

**Comment by morris** [comment]
Yes it is, there's even a glob method, just didn't get around to add the functioniality to rmdir/delete. They do need some specialized behavior though. For example `rmdir( 'dir/**', cb )` should just behave as `rmdir( 'dir', cb )` and certainly not glob anything. It's probably not that hard but it'll need some thought :)
